plat/kvm/x86/acpi.c

```diff
@@ -0,0 +1,191 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/*
+ * Authors: Cristian Vijelie <cristianvijelie@gmail.com>
+ *
+ * Copyright (c) 2021, University Politehnica of Bucharest. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. Neither the name of the copyright holder nor the names of its
+ *    contributors may be used to endorse or promote products derived from
+ *    this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#include <uk/print.h>
+#include <string.h>
+#include <x86/acpi/acpi.h>
+#include <x86/acpi/madt.h>
+#include <uk/assert.h>
+
+__u8 acpi_version;
+struct RSDPDescriptor *rsdp;
+struct RSDPDescriptor20 *rsdp2;
+struct RSDT *rsdt;
+struct XSDT *xsdt;
+struct MADT *madt;
+
+static inline __u8 detect_acpi_version(void *ptr)
+{
+	struct RSDPDescriptor *rsdp = (struct RSDPDescriptor *)ptr;
+
+	UK_ASSERT(ptr);
+
+	return rsdp->Revision;
+}
+
+static inline int rsdp_checksum(__u64 *ptr)
```

Comment:
Somehow, Simon comments got lost in my github for this function. So in case my addition is missing. I agree with Simon, but would use the term `verify`. I would rewrite the function as:
```C
static inline int verify_rsdp_checksum(struct RSDPDescriptor *rdsp)
{
	__u8 checksum = 0;
	__u8 *ptr = (__u8 *)rdsp;

	UK_ASSERT(rdsp);

	while (ptr < (__u8 *)(rdsp + 1))
		checksum += *ptr++;

	return (checksum == 0) ? 0 : -1;
}
```